Enable cpu-info module for sp7 platforms

abinayaddhandapani opened the pull request:
- Converted the build system from CMake to Meson.
- Added clang-formatter and formatted the codebase using clang-format-18
- Created object paths /xyz/openbmc_project/inventory/ system/processor/P0 and /xyz/openbmc_project/inventory/ system/processor/P1 based on the socket count read from the board ID.

ojayanth commented:
Abinya, 
- Good to see Meson migration, which will help build time. Better to Keep cmake build support some time to help users to migrate to Meson based recipe.
- Split the commits based on the PR description (3 commits )
- Include clang version used to format .

abinayaddhandapani commented:
@ojayanth  Some of the recipes like amd-ras , amd-mdc-sdu , bmc-ppr are already migrated to meson.build.
So as part of enabling cpu-info module in integ_sp7 branch , this change is added.
If the user needs reference code for CMake build , it can be referred from the main branch.

The clang-formatter that is used to format the source code is already added to the PR.
https://github.com/AMDESE/bmc-cpuinfo/pull/12/commits/b0c40d262be752c2914765a07f0a11e73c815893#diff-1026e0038b722990204a42bed8a6f7c0ec2302aa79e3fad1959d62ba968edfa2


Pushed the changes to split the PR into 3 different commits.

Commit:
clang-format: Add .clang-format file

clang-format-18 for each input file will try to
find the .clang-format file located in the closest
parent directory of the input file. Since the C++ code is
in bmc-cpuinfo directory it has a specific .clang-format.

Signed-off-by: Abinaya Dhandapani <abinaya.dhandapani@amd.com>

Commit:
CMake to Meson conversion

Converted the build system from CMake to Meson.

Signed-off-by: Abinaya Dhandapani <abinaya.dhandapani@amd.com>

Commit:
Enable cpu-info module for sp7 platforms

Enabling this module for integ_sp7 branch requires
creation of object paths /xyz/openbmc_project/inventory/
system/processor/P0 and /xyz/openbmc_project/inventory/
system/processor/P1 based on the socket count read
from the board ID.

Signed-off-by: Abinaya Dhandapani <abinaya.dhandapani@amd.com>
